paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -1037,33 +1031,43 @@ def on_enter_deploy_cancelled(self) -> None:
         if self.deploy_group_is_set_to_notify("notify_after_abort"):
             self.ping_authors("Deploy cancelled")
 
-    def do_wait_for_deployment(self, target_commit: str) -> None:
+    def stop_waiting_for_deployment(self, target_commit: str) -> None:
+        try:
+            self.wait_for_deployment_tasks[target_commit].cancel()
+        except KeyError:
+            pass
```

Comment:
Theoretically, you could also run into an `asyncio.InvalidStateError` if you attempted to cancel on a non-pending task, so I'd catch that exc too. For good practice, I'd also remove the cancelled task from the tasks dict, so you'd never run into that case.

nit: I'd also await the task afterwards to allow it to finish gracefully cancelling, but right now that would just re-raise the `CancelledError` that killed that task in this control task, which would stop the entire thing, so it's not really necessary. I'd actually prefer that the commit's task just `pass`ed on a cancel, but that's just my preference. Feel free to ignore.